content/blog/2020/08/2020-08-24-custom-distribution-service.adoc

```diff
@@ -0,0 +1,234 @@
+---
+
+
+Hello everyone,
+
+This is the final blog post for the Custom Distribution Service project during the Google Summer of Code timeline .I have mixed feelings since we are almost near the finish line for one of the most amazing open source programs out there. However it is time to wrap things up for this project and achieve a state where the project can be built upon and extended further. This phase has been super busy with respect to the bug fixes, testing and getting the project hosted, so let us get straight into the updates.
+
+== Fixes and Code quality assurance
+
+=== Set Jenkinsfile agent to linux
+
+We realised that the build was failing on windows and that there was not really a use-case
+for running it on windows as of maybe. Maybe it could be on a future roadmap. Therefore we 
+decided to shift the testing to only linux agents
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/116[#116]
+
+=== Backend port error message
+
+Spring boot has a default message on the `port:8080` and therefore we wanted to change 
+it to a custom message on the backend. So the major takeaway here is that we needed to 
+implement the `Error Controller` interface and include a custom message in it. 
+This was technical debt from the last phase and was completed and merged during this phase.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/92[#92]
+
+=== PMD Analysis
+
+This was suggested by Martin which was a real life-saver. The project originally did not 
+include PMD and was added this phase. It helped me catch tons of errors. When the initial 
+PMD check was run and we found approximately 162 PMD errors. We realised some of them were not
+relevant and some of them could be fixed later. 
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/102[#102]
+
+=== Findbugs Analysis
+
+Another tool to improve code quality that we included in this phase was findbugs. 
+It did catch around 5-10 bugs in my code which I immediately resolved. Most of them were 
+around the `Closeable HTTP Request` and an easy fix was the `try with resources`.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/118[#118]
+
+=== Jacoco Code Coverage
+
+We needed to make sure most of the code we write had proper coverage for all branches and
+lines. Therefore we decided to include a JaCoco Code Coverage reporter that helped us find the
+uncovered lines and areas we need to improve coverage on.
+
+* Pull Request link:https://github.com/jenkinsci/custom-distribution-service/pull/103[#103]
+
+=== Remove JCasC generation
+
+While developing the service we quickly realised that the generation of the war package broke if we 
+included a `configuration as code` section but did not provide a path to the corresponding required `yml`
+file. Therefore we took a decision to remove the `casc` section all together. Maybe it will comeback in
+a future patch
```

Comment:
You can add a link to the GitHub issue to add it back so if someone is interested, they can follow that ticket.